scanner/src/main/kotlin/storages/ClearlyDefinedStorage.kt

```diff
@@ -141,51 +142,62 @@ class ClearlyDefinedStorage(
      * using the given [coordinates] for looking up the data. These may not necessarily be equivalent to the
      * identifier, as we try to lookup source code results if a GitHub repository is known.
      */
-    private fun readFromClearlyDefined(
+    private suspend fun readFromClearlyDefined(
         id: Identifier,
         coordinates: ClearlyDefinedService.Coordinates
     ): Result<ScanResultContainer> {
         val startTime = Instant.now()
         log.info { "Looking up results for '${id.toCoordinates()}'." }
 
+        @Suppress("TooGenericExceptionCaught")
         return try {
-            val tools = execute(
-                clearlyDefinedService.harvestTools(
-                    coordinates.type,
-                    coordinates.provider,
-                    coordinates.namespace.orEmpty(),
-                    coordinates.name,
-                    coordinates.revision.orEmpty()
-                )
+            val tools = clearlyDefinedService.harvestTools(
+                coordinates.type,
+                coordinates.provider,
+                coordinates.namespace.orEmpty(),
+                coordinates.name,
+                coordinates.revision.orEmpty()
             )
 
             findScanCodeVersion(tools, coordinates)?.let { version ->
                 loadScanCodeResults(id, coordinates, version, startTime)
             } ?: emptyResult(id)
-        } catch (e: IOException) {
-            e.showStackTrace()
+        } catch (e: HttpException) {
+            e.response()?.errorBody()?.string()?.let {
+                log.error { "Error response from ClearlyDefined is: $it" }
+            }
+
+            handleException(id, e)
+        } catch (e: Exception) {
```

Comment:
Can we limit this to `IOException` (`ConnectException` at least is one) to avoid the annotation in line 152?

Reply:
Unfortunately not; it is a `java.net.ConnectException`. I think, in total 3 catch clauses are required for `IOException`, `HttpConnection`, and `ConnectException`. Do you prefer to list them explicitly?

Reply:
> Unfortunately not; it is a `java.net.ConnectException`

But that *is* an `IOException` according to https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/net/ConnectException.html, or am I missing something?

> Do you prefer to list them explicitly?

No 😁 I wish Kotlin would finally support [multi-catch](https://youtrack.jetbrains.com/issue/KT-7128).

Reply:
Ah, you are right, wired hierarchy of inheritance. Changed to `IOException`.